fastchat/serve/monitor/elo_analysis.py

```diff
@@ -498,12 +498,20 @@ def construct_style_matrices(
 def get_bootstrap_result_style_control(X, Y, models, func_compute_elo, num_round=1000):
     elos = []
     coefs = []
+    assert X.shape[0] % 2 == 0 and X.shape[0] == Y.shape[0]
+    half_X = X[
+        : int(X.shape[0] / 2)
+    ]  # Since we duplicate the battles when constructing X and Y, we don't want to sample the duplicates
+    half_Y = Y[: int(Y.shape[0] / 2)]
+
     for _ in tqdm(range(num_round), desc="bootstrap"):
         indices = np.random.choice(
-            list(range(len(battles))), size=(len(battles)), replace=True
+            list(range(half_X.shape[0])), size=(half_X.shape[0]), replace=True
         )
-        _X = X[indices]
-        _Y = Y[indices]
+        _X = np.concatenate([half_X[indices], half_X[indices]])
+        _Y = np.concatenate([half_Y[indices], half_Y[indices]])
+        assert _X.shape == X.shape and _Y.shape == Y.shape
```

Comment:
do we need to construct `half_X`? or can we just do 

```
len = int(X.shape[0] / 2)
indices = np.random.choice(
    list(range(len)), size=len, replace=True
)
_X = np.concatenate([X[indices], X[indices]])
```

Reply:
Lowkey you are so right!